sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/windowExpressions.scala

```diff
@@ -363,6 +363,11 @@ abstract class OffsetWindowFunction
    */
   val direction: SortDirection
 
+  /**
+   * Whether the offset is based on the entire frame.
+   */
+  val isWholeBased: Boolean = false
```

Comment:
Hm, quick question, why didn't you just pattern match at `WindowExecBase.scala ` instead of having this flag?

Reply:
I added this flag used to distinguish `OffsetWindowFunctionFrame` and `FixedOffsetWindowFunctionFrame`.

Reply:
Can't you just pattern match at https://github.com/apache/spark/pull/28685/files/8a8ec047c140e2329d4a6eac82364274535310fb#diff-f3219c8c886b2e790ef0716b1298d7d0R140?

Reply:
@HyukjinKwon Thanks for your review. But I can't understand your means. Could you tell me more?